circuits/cpp/src/aztec3/circuits/rollup/test_utils/utils.cpp

```diff
@@ -130,6 +138,62 @@ BaseRollupInputs base_rollup_inputs_from_kernels(std::array<KernelData, 2> kerne
     baseRollupInputs.new_commitments_subtree_sibling_path =
         get_sibling_path<PRIVATE_DATA_SUBTREE_INCLUSION_CHECK_DEPTH>(private_data_tree, 0, PRIVATE_DATA_SUBTREE_DEPTH);
 
+    // Update public data tree to generate sibling paths: we first set the initial public data tree to the result of all
+    // state reads and old_values from state transitions. Note that, if the right tx reads or writes an index that was
+    // already processed by the left one, we don't want to reflect that as part of the initial state, so we skip those.
+    std::set<uint256_t> visited_indices;
+    for (size_t i = 0; i < 2; i++) {
+        for (auto state_read : kernel_data[i].public_inputs.end.state_reads) {
+            auto leaf_index = uint256_t(state_read.leaf_index);
+            if (state_read.is_empty() || visited_indices.contains(leaf_index))
+                continue;
+            visited_indices.insert(leaf_index);
+            public_data_tree.update_element(leaf_index, state_read.value);
+        }
+
+        for (auto state_write : kernel_data[i].public_inputs.end.state_transitions) {
+            auto leaf_index = uint256_t(state_write.leaf_index);
+            if (state_write.is_empty() || visited_indices.contains(leaf_index))
+                continue;
+            visited_indices.insert(leaf_index);
+            public_data_tree.update_element(leaf_index, state_write.old_value);
+        }
+    }
+
+    baseRollupInputs.start_public_data_tree_snapshot = {
+        .root = public_data_tree.root(),
+        .next_available_leaf_index = 0,
+    };
+
+    // Then we collect all sibling paths for the reads in the left tx, and then apply the state transitions while
+    // collecting their paths. And then repeat for the right tx.
+    for (size_t i = 0; i < 2; i++) {
+        for (size_t j = 0; j < STATE_READS_LENGTH; j++) {
+            auto state_read = kernel_data[i].public_inputs.end.state_reads[j];
+            if (state_read.is_empty())
+                continue;
+            auto leaf_index = size_t(uint256_t(state_read.leaf_index));
```

Comment:
Leave as `uint256_t`?

Reply:
Damn, good catch! I searched for other size_t casts and could find any others in the diff, so we should be good.